crates/npmrc/src/custom_deserializer.rs

```diff
@@ -16,25 +16,35 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
     vec!["*eslint*".to_string(), "*prettier*".to_string()]
 }
 
+pub fn is_home(path: String) -> bool {
+    path.starts_with("~/") || path.starts_with("~\\")
+}
+
+pub fn resolve_store_dir(pacquet_home: String) -> PathBuf {
+    if is_home(pacquet_home.clone()) {
+        let home_dir = dirs::home_dir().expect("Home directory is not available");
+        return home_dir.join(&pacquet_home[2..]);
+    }
+    PathBuf::from(pacquet_home.as_str())
+}
+
 /// If the $PACQUET_HOME env variable is set, then $PACQUET_HOME/store
 /// If the $XDG_DATA_HOME env variable is set, then $XDG_DATA_HOME/pacquet/store
 /// On Windows: ~/AppData/Local/pacquet/store
 /// On macOS: ~/Library/pacquet/store
 /// On Linux: ~/.local/share/pacquet/store
 pub fn default_store_dir() -> PathBuf {
-    // TODO: If env variables start with ~, make sure to resolve it into home_dir.
+    let home_dir = dirs::home_dir().expect("Home directory is not available");
+
+    // just home need to support resolve ~ prefix
     if let Ok(pacquet_home) = env::var("PACQUET_HOME") {
-        return PathBuf::from(pacquet_home).join("store");
+        return resolve_store_dir(pacquet_home).join("store");
     }
 
     if let Ok(xdg_data_home) = env::var("XDG_DATA_HOME") {
         return PathBuf::from(xdg_data_home).join("pacquet/store");
     }
 
-    // Using ~ (tilde) for defining home path is not supported in Rust and
-    // needs to be resolved into an absolute path.
-    let home_dir = home::home_dir().expect("Home directory is not available");
-
```

Comment:
`default_store_dir` still does too many thing. I want `resolve_store_dir` to do most thing and the only thing `default_store_dir` should do is passing variables to `resolve_store_dir`.